From: Lucas Kanashiro <kanashiro@ubuntu.com>
Date: Wed, 21 Jun 2023 16:35:33 -0300
Subject: Skip tests which require a certain level of privilege

Forwarded: not-needed
Last-Updated: 2024-06-04

During build we cannot bindmount sysfs and cgroupfs in a chroot which leads to
failures.
---
 internal/cri/server/container_create_linux_test.go |  13 +
 .../container_update_resources_linux_test.go       |   1 +
 pkg/cri/server/container_create_test.go            | 467 +++++++++++++++++++++
 3 files changed, 481 insertions(+)
 create mode 100644 pkg/cri/server/container_create_test.go

diff --git a/internal/cri/server/container_create_linux_test.go b/internal/cri/server/container_create_linux_test.go
index 521d795..4d86d61 100644
--- a/internal/cri/server/container_create_linux_test.go
+++ b/internal/cri/server/container_create_linux_test.go
@@ -194,6 +194,7 @@ func getCreateContainerTestData() (*runtime.ContainerConfig, *runtime.PodSandbox
 }
 
 func TestContainerCapabilities(t *testing.T) {
+	t.Skip("It requires HugeTLB controller enabled which requires mounting cgroupfs. Not achievable during the build.")
 	testID := "test-id"
 	testSandboxID := "sandbox-id"
 	testContainerName := "container-name"
@@ -281,6 +282,7 @@ func TestContainerCapabilities(t *testing.T) {
 }
 
 func TestContainerSpecTty(t *testing.T) {
+	t.Skip("It requires HugeTLB controller enabled which requires mounting cgroupfs. Not achievable during the build.")
 	testID := "test-id"
 	testSandboxID := "sandbox-id"
 	testContainerName := "container-name"
@@ -303,6 +305,7 @@ func TestContainerSpecTty(t *testing.T) {
 }
 
 func TestContainerSpecDefaultPath(t *testing.T) {
+	t.Skip("It requires HugeTLB controller enabled which requires mounting cgroupfs. Not achievable during the build.")
 	testID := "test-id"
 	testSandboxID := "sandbox-id"
 	testContainerName := "container-name"
@@ -325,6 +328,7 @@ func TestContainerSpecDefaultPath(t *testing.T) {
 }
 
 func TestContainerSpecReadonlyRootfs(t *testing.T) {
+	t.Skip("It requires HugeTLB controller enabled which requires mounting cgroupfs. Not achievable during the build.")
 	testID := "test-id"
 	testSandboxID := "sandbox-id"
 	testContainerName := "container-name"
@@ -342,6 +346,7 @@ func TestContainerSpecReadonlyRootfs(t *testing.T) {
 }
 
 func TestContainerSpecWithExtraMounts(t *testing.T) {
+	t.Skip("It requires HugeTLB controller enabled which requires mounting cgroupfs. Not achievable during the build.")
 	testID := "test-id"
 	testSandboxID := "sandbox-id"
 	testContainerName := "container-name"
@@ -391,6 +396,7 @@ func TestContainerSpecWithExtraMounts(t *testing.T) {
 }
 
 func TestContainerAndSandboxPrivileged(t *testing.T) {
+	t.Skip("It requires HugeTLB controller enabled which requires mounting cgroupfs. Not achievable during the build.")
 	testID := "test-id"
 	testSandboxID := "sandbox-id"
 	testContainerName := "container-name"
@@ -446,6 +452,7 @@ func TestContainerAndSandboxPrivileged(t *testing.T) {
 }
 
 func TestPrivilegedBindMount(t *testing.T) {
+	t.Skip("It requires privilege to mount sysfs and cgroupfs. Not achievable during the build.")
 	testPid := uint32(1234)
 	c := newTestCRIService()
 	testSandboxID := "sandbox-id"
@@ -609,6 +616,7 @@ func TestMountPropagation(t *testing.T) {
 }
 
 func TestPidNamespace(t *testing.T) {
+	t.Skip("It requires HugeTLB controller enabled which requires mounting cgroupfs. Not achievable during the build.")
 	testID := "test-id"
 	testPid := uint32(1234)
 	testSandboxID := "sandbox-id"
@@ -845,6 +853,7 @@ func TestUserNamespace(t *testing.T) {
 }
 
 func TestNoDefaultRunMount(t *testing.T) {
+	t.Skip("It requires HugeTLB controller enabled which requires mounting cgroupfs. Not achievable during the build.")
 	testID := "test-id"
 	testPid := uint32(1234)
 	testSandboxID := "sandbox-id"
@@ -1196,6 +1205,7 @@ func TestGenerateApparmorSpecOpts(t *testing.T) {
 }
 
 func TestMaskedAndReadonlyPaths(t *testing.T) {
+	t.Skip("It requires HugeTLB controller enabled which requires mounting cgroupfs. Not achievable during the build.")
 	testID := "test-id"
 	testSandboxID := "sandbox-id"
 	testContainerName := "container-name"
@@ -1294,6 +1304,7 @@ func TestMaskedAndReadonlyPaths(t *testing.T) {
 }
 
 func TestHostname(t *testing.T) {
+	t.Skip("It requires HugeTLB controller enabled which requires mounting cgroupfs. Not achievable during the build.")
 	testID := "test-id"
 	testSandboxID := "sandbox-id"
 	testContainerName := "container-name"
@@ -1589,6 +1600,7 @@ func TestNonRootUserAndDevices(t *testing.T) {
 }
 
 func TestPrivilegedDevices(t *testing.T) {
+	t.Skip("It requires privilege to test devices. Not achievable during the build.")
 	testPid := uint32(1234)
 	c := newTestCRIService()
 	testSandboxID := "sandbox-id"
@@ -1680,6 +1692,7 @@ func TestPrivilegedDevices(t *testing.T) {
 }
 
 func TestBaseOCISpec(t *testing.T) {
+	t.Skip("It requires HugeTLB controller enabled which requires mounting cgroupfs. Not achievable during the build.")
 	baseLimit := int64(100)
 	c := newTestCRIService(withRuntimeService(&fakeRuntimeService{
 		ocispecs: map[string]*oci.Spec{
diff --git a/internal/cri/server/container_update_resources_linux_test.go b/internal/cri/server/container_update_resources_linux_test.go
index 1e3132b..1d27487 100644
--- a/internal/cri/server/container_update_resources_linux_test.go
+++ b/internal/cri/server/container_update_resources_linux_test.go
@@ -30,6 +30,7 @@ import (
 )
 
 func TestUpdateOCILinuxResource(t *testing.T) {
+	t.Skip("It requires some privileges not achievable during the build.")
 	oomscoreadj := new(int)
 	*oomscoreadj = -500
 	expectedSwap := func(swap int64) *int64 {
